qiskit/algorithms/minimum_eigen_solvers/vqe.py

```diff
@@ -402,19 +335,86 @@ def extract_circuits(op):
     def supports_aux_operators(cls) -> bool:
         return True
 
-    def _run(self) -> 'VQEResult':
-        """Run the algorithm to compute the minimum eigenvalue.
+    def _eval_aux_ops(self,
+                      aux_operators: List[OperatorBase],
+                      threshold: float = 1e-12) -> None:
+        # Create new CircuitSampler to avoid breaking existing one's caches.
+        sampler = CircuitSampler(self.quantum_instance)
 
-        Returns:
-            The result of the VQE algorithm as ``VQEResult``.
+        aux_op_meas = self.expectation.convert(StateFn(ListOp(aux_operators),
+                                                       is_measurement=True))
+        aux_op_expect = aux_op_meas.compose(CircuitStateFn(self.get_optimal_circuit()))
+        values = np.real(sampler.convert(aux_op_expect).eval())
 
-        Raises:
-            AlgorithmError: Wrong setting of operator and backend.
-        """
-        if self.operator is None:
+        # Discard values below threshold
+        aux_op_results = (values * (np.abs(values) > threshold))
+        # Deal with the aux_op behavior where there can be Nones or Zero qubit Paulis in the list
+        _aux_op_nones = [op is None for op in aux_operators]
+        self._ret['aux_ops'] = [None if is_none else [result]
+                                for (is_none, result) in zip(_aux_op_nones, aux_op_results)]
+        # As this has mixed types, since it can included None, it needs to explicitly pass object
+        # data type to avoid numpy 1.19 warning message about implicit conversion being deprecated
+        self._ret['aux_ops'] = np.array([self._ret['aux_ops']], dtype=object)
+
+    def _check_operator(self, operator: Union[OperatorBase, LegacyBaseOperator]) -> OperatorBase:
+        """ set operator """
+        if isinstance(operator, LegacyBaseOperator):
+            operator = operator.to_opflow()
+        self._expect_op = None
+        self._check_operator_varform(operator)
+        # Expectation was not passed by user, try to create one
+        if not self._user_valid_expectation:
+            self._try_set_expectation_value_from_factory(operator)
+        return operator
+
+    def _check_aux_operators(self,
+                             operator: Union[OperatorBase, LegacyBaseOperator],
+                             aux_operators: Optional[
+                                 Union[OperatorBase,
+                                       LegacyBaseOperator,
+                                       List[Optional[Union[OperatorBase,
+                                                           LegacyBaseOperator]]]]]) \
+            -> List[OperatorBase]:
+        """ Set aux operators """
```

Comment:
Is this docstring correct?

Reply:
No, but I removed and refactored this method now.